drivers/clock_control/clock_stm32_mco.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2024 Benjamin Björnsson <benjamin.bjornsson@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_mco_clock
+
+#include <zephyr/arch/common/sys_bitops.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/pinctrl.h>
+
+#include <stdint.h>
+
+struct clock_control_stm32_mco_config {
+	uint32_t base;
+	const struct stm32_pclken *pclken;
+	size_t pclk_len;
+	const struct pinctrl_dev_config *pcfg;
+};
+
+static int clock_control_stm32_mco_init(const struct device *dev)
+{
+	const struct clock_control_stm32_mco_config *config = dev->config;
+
+	for (int i = 0; i < config->pclk_len; i++) {
+		sys_clear_bits(config->base + STM32_CLOCK_REG_GET(config->pclken->enr),
+			       STM32_CLOCK_MASK_GET(config->pclken[i].enr)
+				       << STM32_CLOCK_SHIFT_GET(config->pclken[i].enr));
+		sys_set_bits(config->base + STM32_CLOCK_REG_GET(config->pclken->enr),
+			     STM32_CLOCK_VAL_GET(config->pclken[i].enr)
+				     << STM32_CLOCK_SHIFT_GET(config->pclken[i].enr));
+	}
```

Comment:
Altenatively, this could remain a routine of the clock_control diver (called before clock configuretion) and as such, reuse `stm32_clock_control_on` internal calls.